meeko/molsetup.py

```diff
@@ -4,623 +4,451 @@
-                name=atom_name,
-                resName=resn,
-                resNum=resi,
-                chain=chain
-            )
-            pdbinfos.append(pdbinfo)
-
-        atom_params = {}
-        atom_counter = 0
-        all_ok = True
-        all_err = ""
-        for (res, atom_names) in zip(res_list, atom_names_list):
-            p, ok, err = PDBQTReceptor.get_params_for_residue(res, atom_names)
-            nr_params_to_add = set([len(values) for _, values in p.items()])
-            if len(nr_params_to_add) != 1:
-                raise RuntimeError("inconsistent number of parameters in %s" % p)
-            nr_params_to_add = nr_params_to_add.pop()
-            all_ok &= ok
-            all_err += err
-            for key in p:
-                atom_params.setdefault(key, [None]*atom_counter)
-                atom_params[key].extend(p[key])
-            atom_counter += nr_params_to_add
-            for key in atom_params:
-                if key not in p:
-                    atom_params[key].extend([None]*nr_params_to_add)
-    
-        if not all_ok:
-            raise RuntimeError(all_err)
-        
-        molsetup = cls()
-        charges = atom_params.pop("gasteiger")
-        atypes = atom_params.pop("atom_types")
-        for i in range(len(x)):
-            molsetup.add_atom(
-                i,
-                coord = (x[i], y[i], z[i]),
-                charge = charges[i],
-                atom_type = atypes[i],
-                element = None,
-                pdbinfo = pdbinfo,
-                chiral = False,
-                ignore = False,
-            )
+        # If atom index is not specified, appends the new atom to the end of the current atom list
+        if atom_index is None:
+            atom_index = len(self.atoms)
+
+        # Inserts dummy atoms if a specified atom index is greater than the current length of the atom list
+        while atom_index > len(self.atoms):
+            self.atoms.append(Atom(len(self.atoms), is_dummy=True))
+
+        # Creates and adds new atom to the atom list
+        new_atom = Atom(
+            atom_index, pdbinfo, charge, element, atom_type, is_ignore, is_chiral, graph
+        )
+        if atom_index < len(self.atoms):
+            self.atoms[atom_index] = new_atom
+            return
+        self.atoms.append(new_atom)
+        return
 
-        molsetup.atom_params = atom_params
+    def add_pseudo_atom(
+        self,
+        pdbinfo: str = DEFAULT_PDBINFO,
+        charge: float = DEFAULT_CHARGE,
+        atom_type: str = DEFAULT_ATOM_TYPE,
+        is_ignore: bool = DEFAULT_IS_IGNORE,
+        anchor_list: list[int] = None,
+        rotatable: bool = False,
+        directional_vectors: list[int] = None,
+    ):
+        """
 
-        return molsetup 
+        Parameters
+        ----------
+        pdbinfo
+        charge
+        atom_type
+        is_ignore
+        anchor_list
+        rotatable
+        directional_vectors
```

Comment:
I was having a hard time seeing where directional vectors were being used? Do we still need them, and if so, what should be the supported datatypes and actions for them?

Reply:
Yep, we don't need them.